extensions/notebook/src/test/common/notebookUtils.test.ts

```diff
@@ -80,4 +86,117 @@ describe('notebookUtils Tests', function (): void {
 			apiWrapperMock.verify(x => x.showErrorMessage(TypeMoq.It.isAny()), TypeMoq.Times.once());
 		});
 	});
+
+	describe('runActiveCell', function () {
+		it('shows error when no notebook visible', async function (): Promise<void> {
+			apiWrapperMock.setup(x => x.getActiveNotebookEditor()).returns(() => undefined);
+			await notebookUtils.runActiveCell();
+			apiWrapperMock.verify(x => x.showErrorMessage(TypeMoq.It.isAny()), TypeMoq.Times.once());
+		});
+
+		it('does not show error when notebook visible', async function (): Promise<void> {
+			let mockNotebookEditor = TypeMoq.Mock.ofType<azdata.nb.NotebookEditor>();
+			apiWrapperMock.setup(x => x.getActiveNotebookEditor()).returns(() => mockNotebookEditor.object);
```

Comment:
I don't really have a huge issue with this in the end - but for something like this there's really no reason why we should be needing to mock out the notebook creation - you can create and run notebooks just fine when running as a test. And testing the actual extension APIs isn't the worst thing (I've found a couple bugs that way) since that's not something that usually gets tested from the core side currently.

Reply:
The reason why I didn't do this was because, to actually run cells, the machine running the tests needs Python + the kernels installed.